File: 0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        if (chars.size() == 1){
            return chars.size();
        }
        string s;
        char prev = chars[0];
        int count = 1;
        for (int i = 1; i < chars.size(); ++i){
            if (chars[i] == prev){
                ++count;
            } else{
                s += prev;
                if (count > 1){
                    s += to_string(count);
                }
                prev = chars[i];
                count = 1;
            }
        }
        s += prev;
        if (count > 1){
            s += to_string(count);
        }
        chars.clear();
        for (auto c: s){
            chars.push_back(c);
        }
        return chars.size();
    }
};

File: 0443-string-compression/0443-string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        check = 0
        while check < len(chars):
            length = check + 1
            count = 1
            while length < len(chars) and chars[length] == chars[check]:
                count += 1
                chars.pop(length)
            
            if 1 < count < 10:
                chars.insert(length, str(count))
                check = length + 1
            elif count >= 10:
                res = count
                check = length
                while res > 0:
                    chars.insert(length, str(res % 10))
                    res = res // 10
                    check += 1
            else:
                check = length
        
        return len(chars)
            